autopush-common/src/db/mod.rs

```diff
@@ -89,12 +81,7 @@ impl StorageType {
         }
         let dsn = dsn
             .clone()
-            .unwrap_or(std::env::var("AWS_LOCAL_DYNAMODB").unwrap_or_default());
-        #[cfg(feature = "dynamodb")]
-        if dsn.starts_with("http") {
-            trace!("Found http");
-            return Self::DynamoDb;
-        }
+            .unwrap_or(std::env::var("GOOGLE_APPLICATION_CREDENTIALS").unwrap_or_default());
```

Comment:
No, the `GOOGLE_APPLICATION_CREDENTIALS` is a file holding the connection credentials, not the DSN to the database. We'd still need that DSN value. (`AWS_LOCAL_DYNAMODB` was a DSN to the locally running database).

If the DSN is not set, we should fail out with a config error.

Reply:
Thanks for the clarification. Ah, so before it was returning an `Option`, so are you saying I should change the function to return a `Result`? I had looked at just calling `expect()` or `unwrap()` at the end of that chain.